sign/sign.go

```diff
@@ -108,15 +108,15 @@ func (s *Signer) SignImage(reference string) (*SignedObject, error) {
 	defer cancel()
 
 	err := s.impl.SignImageInternal(ctx, ko, regOpts,
-		s.options.Annotations, imgs, "", true, outputSignature,
+		s.options.Annotations, images, "", true, outputSignature,
 		outputCertificate, "", true, false, "")
 	if err != nil {
 		return nil, errors.Wrapf(err, "sign reference: %s", reference)
 	}
 
-	object, err := s.impl.VerifyImageInternal(s, reference)
+	object, err := s.impl.VerifyImageInternal(ctx, s.options.PrivateKeyPath, images)
```

Comment:
I think we should use the public key here:

```suggestion
	object, err := s.impl.VerifyImageInternal(ctx, s.options.PublicKeyPath, images)
```

Reply:
Good catch